backend/de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5565380_sys_gh3856_setM_Forecast.DatePromissed_IfNotExists.sql

```diff
@@ -0,0 +1,18 @@
+UPDATE m_forecast
+SET datepromised = (
+    SELECT startdate::Timestamp
+    FROM c_period p
+    WHERE p.c_period_id = m_forecast.c_period_id
+
+    UNION
+
+    SELECT to_timestamp(fiscalyear, 'YYYY') /* eg. 2020.01.01 */
+    FROM c_year y
+    WHERE y.c_year_id = m_forecast.m_forecast_id
+
+    LIMIT 1
```

Comment:
u think you need to do sth like
```suggestion
    SELECT 1 as seq, startdate::Timestamp
    FROM c_period p
    WHERE p.c_period_id = m_forecast.c_period_id

    UNION

    SELECT 2, to_timestamp(fiscalyear, 'YYYY') /* eg. 2020.01.01 */
    FROM c_year y
    WHERE y.c_year_id = m_forecast.m_forecast_id <-- this seems wrong
    
    UNION 
    
    SELECT 3, '2010-01-01'
    
    ORDER BY seq
    LIMIT 1
```

* to make sure your subselect prefers the ones with a period_id
* m_forecast.c_year_id is not mandatory, so AFAIU we need a 3rd fallback

Reply:
Fixed